homeassistant/helpers/selector.py

```diff
@@ -1063,9 +1064,9 @@ class NumberSelector(Selector[NumberSelectorConfig]):
                 vol.Optional("min"): vol.Coerce(float),
                 vol.Optional("max"): vol.Coerce(float),
                 # Controls slider steps, and up/down keyboard binding for the box
-                # user input is not rounded
-                vol.Optional("step", default=1): vol.Any(
-                    "any", vol.All(vol.Coerce(float), vol.Range(min=1e-3))
+                # if step is an integer, a validation of integer will be enforced
+                vol.Optional("step", default=1.0): vol.Any(
+                    "any", int, vol.All(vol.Coerce(float), vol.Range(min=1e-3))
```

Comment:
Don't we need `vol.Range(min=1))` also for int so it's not set to `0`?

Reply:
Makes sense 👍 
Added...